### PR TITLE
samples: add samples from pubsub/cloud-client

### LINKED ISSUE #136: Add pubsub/cloud-client samples from the samples repo
A transition to the new code generator (#131) requires the code samples to be moved from the common samples repo to here.

### PR DESCRIPTION
**Pull request by plamut**
Samples from https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/pubsub/cloud-client

Closes #136.
Towards #131.

### CODE REVIEW

**Comment by plamut** [comment]
Somebody needs to override the CLA issue. It is _probably_ fine, as the commits were just taken from [another repo](https://github.com/GoogleCloudPlatform/python-docs-samples/) under our control, but I cannot verify that 100%.

**Comment by busunkim96** [comment]
Files are being moved from a different repo and already have signed CLAs.

**Comment by busunkim96** [comment]
Just opened a PR internally to add the additional Kokoro checks for samples.

**Comment by anguillanneuf** [comment]
@plamut Let's coordinate on the merge because we don't want Cloud docs to link to non-existing samples. I will prepare a CL for docs so we can merge/submit at roughly the same time.

**Comment by plamut** [comment]
@anguillanneuf Generated/updated the templates, including the `noxfile.py` for the snippets. The PR is ready for another round of reviews now.

Re: internal PR - I _think_ it should be fine (can't see its contents), as the samples have been copied from the samples repo. When this is approved an merged, we can just "redirect" the docs to point at the new location.

**Comment by plamut** [comment]
@busunkim96 @anguillanneuf Uh-oh, it appears that the sample history was squash-merged instead of using rebase and merge as specified by the migration doc.

How important is the history in this case? Do we need to purge this PR and merge it again with history preserved?

**Comment by anguillanneuf** [comment]
Is it still possible to rebase? If so, please do! Thanks @plamut.

File: samples/snippets/README.rst.in [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    `Google Cloud Pub/Sub`_ is a fully-managed real-time messaging service that
+    allows you to send and receive messages between independent applications.
+
+setup:
+- auth
+- install_deps
+
+samples:
+- name: Quickstart
+  file: quickstart.py
```
Comment (sgerrand) [inline samples/snippets/README.rst.in:17]: It's worth noting that this file doesn't exist.

File: samples/snippets/README.rst [hunk 2]
```diff
@@ -0,0 +1,282 @@
+
+.. This file is automatically generated. Do not edit this file directly.
+
+Google Cloud Pub/Sub Python Samples
+===============================================================================
+
+.. image:: https://gstatic.com/cloudssh/images/open-btn.png
+   :target: https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/python-docs-samples&page=editor&open_in_editor=pubsub/cloud-client/README.rst
```
Comment (sgerrand) [inline samples/snippets/README.rst:8]: This URL should now be https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/googleapis/python-pubsub&page=editor&open_in_editor=samples/snippets/iam.py,samples/snippets/README.rst

File: samples/snippets/README.rst [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+
+
+Install Dependencies
+++++++++++++++++++++
+
+#. Clone python-docs-samples and change directory to the sample directory you want to use.
+
+    .. code-block:: bash
+
+        $ git clone https://github.com/GoogleCloudPlatform/python-docs-samples.git
```
Comment (sgerrand) [inline samples/snippets/README.rst:44]:
```diff
-        $ git clone https://github.com/GoogleCloudPlatform/python-docs-samples.git
+        $ git clone https://github.com/googleapis/python-pubsub.git
```

**Pull request merged**